### DISCUSSION TITLE
Run server without built-in agent?

### DISCUSSION THREAD

**Discussion by skirsten** [k3s-io/k3s#5748] [Q&A] [answered]
Is it possible to run the k3s server without the built-in agent, so it will only host the api-server etc. and will not show up on kubectl?

I found some old discussions on this here https://github.com/k3s-io/k3s/issues/978#issuecomment-552966999 that removed this feature.
Is there any way to achieve this using different flags or does this have to be re-implemented?

**Comment by skirsten** [comment, marked as answer]
Seems like the option still exists but is hidden: https://github.com/k3s-io/k3s/blob/a5414bb1fc904e3d44a6a66ccffa819340e712a8/pkg/cli/cmds/server.go#L528
I will check it out

**Reply by brandond** [reply]
Yep. The option is there but is unsupported/experimental.

**Reply by skirsten** [reply]
Works like a charm :heart: k3s
